lib/services/provisioning.go

```diff
@@ -124,28 +124,35 @@ func MarshalProvisionToken(t ProvisionToken, opts ...MarshalOption) ([]byte, err
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	type token1 interface {
-		V1() *ProvisionTokenV1
-	}
-	type token2 interface {
-		V2() *ProvisionTokenV2
-	}
 
-	version := cfg.GetVersion()
-	switch version {
-	case V1:
+	version := t.GetVersion()
+	// ProvisionTokenV1 doesn't implement ProvisionToken interface,
```

Comment:
This is confusing - if `ProvisionTokenV1` does not implement `ProvisionToken` interface then it can never be passed into `MarshalProvisionToken` as argument? I think the local V1/V2 interfaces are meant to convert the V2 -> V1 when option specifies `V1` as target version. The difference here to the previous implementation is that it add handling of the empty version as a fallback for `V1` - can you explain why this is a special case?